src/async-data/reducer.js

```diff
@@ -0,0 +1,29 @@
+export const createReducer = (defaultState, actions) => {
+	const DEFAULT_STATE = {
+		requesting: false,
+		...defaultState
+	}
+	return (state = DEFAULT_STATE, action) => {
+		switch (action.type) {
+			case actions.REQUESTED:
+			case actions.RETRY:
+				return { ...state, requesting: true, succeeded: false }
```

Comment:
Same comment as before, if state is a deep object spreading will not create an inmutable version of it.